distributed/shuffle/_shuffle_extension.py

```diff
@@ -434,6 +501,7 @@ async def _get_shuffle(
 
                 raise Reschedule()
             else:
+                self.raise_if_closed()
                 if shuffle_id not in self.shuffles:
                     shuffle = Shuffle(
```

Comment:
We just discussed that intstatiating a new shuffle instance (and also talking to the scheduler itself) is an error case that can only be caught by implementing the raise_if_closed pattern on extension level. everything else could be handled on shuffle instance level instead